src/main.py

```diff
@@ -161,16 +167,46 @@ def open_pr(subject, body, branch):
     base = "master"
     head = "{}:{}".format(repo.owner.login, branch)
     pr_message = ((body or "") + "\n\n" + DISCLAIMER).strip()
-    # Include closed PRs – if the maintainer has closed our last PR, we don't want to
-    # open another one.
-    for pr in origin_repo.get_pulls(state="all", base=base, head=head):
+
+    if os.path.isfile("flathub.json"):
+        with open("flathub.json") as f:
+            repocfg = json.load(f)
+            automerge = repocfg.get("automerge-flathubbot-prs")
+
+    # Enable automatic merge if external data is broken unless explicitly disabled
+    if automerge is not False and manifest_checker:
+        for data in manifest_checker.get_outdated_external_data():
+            if data.new_version and data.state == ExternalData.State.BROKEN:
+                automerge = True
+                break
+
+    prs = origin_repo.get_pulls(state="all", base=base, head=head)
+
+    # If the maintainer has closed our last PR or it was merged,
+    # we don't want to open another one.
+    closed_prs = [pr for pr in prs if pr.state == "closed"]
+    for pr in closed_prs:
         log.info(
             "Found existing %s PR: %s",
             "merged" if pr.is_merged() else pr.state,
             pr.html_url,
         )
         return
 
+    if origin_repo.permissions.push:
+        open_prs = [pr for pr in prs if pr.state == "open"]
+        for pr in open_prs:
+            log.info("Found open PR: %s", pr.html_url)
+
+            if automerge:
+                pr_commit = pr.head.repo.get_commit(pr.head.sha)
+                if pr_commit.get_combined_status().state == "success" and pr.mergeable:
+                    log.info("PR passed CI and is mergeable, merging", pr.html_url)
+                    pr.merge(merge_method='rebase')
```

Comment:
It might be nice if the bot left a note on the PR to explain why it is doing this.

Reply:
Done.